tests/integration/test_store_synapse.py

```diff
@@ -125,3 +132,306 @@ def test_process_row_annotations_get_validation(
             "value2",
             "value3",
         ]
+
+    @pytest.mark.parametrize(
```

Comment:
can we also add the situations where no files are expected?

Reply:
@linglp that's a good point, thanks!

Reply:
@linglp so that's actually covered by the case discussed [here](https://github.com/Sage-Bionetworks/schematic/pull/1529#discussion_r1854383443), when there are no files a `LookupError` is raised
cc @BryanFauble

Reply:
Sorry for the late reply. Thanks for pointing out Bryan's comment. I noticed that we have this empty dataset id: syn64319379 in schematic main.Should we use this dataset in the integration test to verify that the error message is returned correctly? 

I tested the API, and it correctly returned the expected error: `LookupError: Dataset syn64319379 could not be found in fileview syn23643253.` However, I think it would be beneficial to include this as a test case in the integration test as well.

Reply:
Thanks @linglp, couple of thoughts here and I'll let the two of you decide if we should add it

1. If we add it, it should be a separate integration test to not complicate the test function with an if/else.
2. As mentioned, there is a test case in test_store.py.  It doesn't hurt to add an integration test, but we would be testing Synapse and the Python client's functionality to return a fileview without that dataset id.  Note, the beauty of the test in `test_store.py` is that it returns a representation of the python client/Synapse should return in the hope that the python client is tested.

Reply:
> Sorry for the late reply. Thanks for pointing out Bryan's comment. I noticed that we have this empty dataset id: syn64319379 in schematic main.Should we use this dataset in the integration test to verify that the error message is returned correctly?

No worries @linglp! That was a new dataset I created to use for a test in response to your comment.

> If we add it, it should be a separate integration test to not complicate this test function with an if/else.

I agree.

> I'll let @GiaJordan do a final call on whether we should or shouldn't add it and why
> As mentioned, there is a test case in test_store.py. It doesn't hurt to add an integration test, but we would be partially testing Synapse and the Python client's functionality to return a fileview without that dataset id. Note, the beauty of the test in test_store.py is that it returns a representation of the python client/Synapse should return in the hope that the python client is tested. That said, there might be a small risk of overwriting the query string for this error to not be raised, but with all the other existing integration tests, it might be difficult to achieve that.

@thomasyu888 I'd prefer to add an integration test. I know there's been focus recently on the speed of the test suite but I think here there's value in ensuring that given known user actions (ie dataset setup) the interactions with synapse are verified to go as we expect.

Reply:
Thanks both!  I'm on board with that ✅

Reply:
Sounds good. And just to note, the 2 new tests take about 11 seconds to run sequentially locally.